src/Altinn.AccessManagement.Persistence/DelegationMetadataRepo.cs

```diff
@@ -0,0 +1,779 @@
+            }
+
+            if (coveredByUserId != null)
+            {
+                query += " AND coveredByUserId = @coveredByUserId";
+                param.Add("coveredByUserId", coveredByUserId);
+            }
+
+            try
+            {
+                var res = await _connection.QueryAsync<DelegationChange>(new CommandDefinition(query, param, cancellationToken: cancellationToken));
+                return res == null ? null : res.ToList();
+            }
+            catch (Exception ex)
+            {
+                activity?.StopWithError(ex);
+                throw;
+            }
+            finally
+            {
+                _connection.Close();
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<DelegationChange>> GetAllCurrentAppDelegationChanges(List<int> offeredByPartyIds, List<string> altinnAppIds = null, List<int> coveredByPartyIds = null, List<int> coveredByUserIds = null, CancellationToken cancellationToken = default)
+        {
+            using var activity = TelemetryConfig.ActivitySource.StartActivity(ActivityKind.Client);
+
+            if (offeredByPartyIds?.Count == 0)
+            {
+                activity?.StopWithError(new ArgumentNullException(nameof(offeredByPartyIds)));
+                throw new ArgumentNullException(nameof(offeredByPartyIds));
+            }
+
+            var param = new Dictionary<string, object>();
+
+            string query = "WITH latestChanges AS (" +
+            "SELECT MAX(delegationChangeId) as latestId" +
+            " FROM delegation.delegationchanges" +
+            " WHERE 1=1";
+
+            if (altinnAppIds?.Count != 0)
+            {
+                query += " AND (altinnAppId IN(@altinnAppIds))";
+                param.Add("altinnAppIds", altinnAppIds);
+            }
+
+            if (offeredByPartyIds?.Count != 0)
+            {
+                query += " AND (offeredByPartyId IN(@offeredByPartyIds))";
+                param.Add("offeredByPartyIds", offeredByPartyIds);
+            }
+
+            if (coveredByPartyIds?.Count != 0)
+            {
+                query += " AND (coveredByPartyId IN(@coveredByPartyIds))";
+                param.Add("coveredByPartyIds", coveredByPartyIds);
+            }
+
+            if (coveredByUserIds?.Count != 0)
+            {
+                query += " AND (coveredByUserId IN(@coveredByUserIds))";
+                param.Add("coveredByUserIds", coveredByUserIds);
+            }
+
+            query += "GROUP BY altinnAppId, offeredByPartyId, coveredByPartyId, coveredByUserId" +
+            ")" + 
+            $"SELECT {defaultColumns}" +
+            " FROM delegation.delegationchanges" +
+            " INNER JOIN latestChanges ON delegationchangeid = latestChanges.latestId";
+
+            try
+            {
+                var res = await _connection.QueryAsync<DelegationChange>(new CommandDefinition(query, param, cancellationToken: cancellationToken));
```

Comment:
If I remember @Alxandr findings when implementing queries in resourceregistry correctly, an importent optimization when moving from functions to queries is to use the prepare statement before execution:
`await cmd.PrepareAsync(cancellationToken);`

Reply:
Yes, I'm testing doing this in the connectionstring. See Line 30 in the same file. Since we have a small number of unique queries this should be more than adequate. This is a candidate for Feature Manager A/B testing in the future to get som stats on the diffrent implementations.